test/TestVectorField.cc

```diff
@@ -0,0 +1,1630 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018, 2019, 2020, 2021 Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#include <gtest/gtest.h>
+
+#include "BackendModel.h"
+#include "CommonTestUtilities.h"
+#include "DataStream/Compression.h"
+#include "DataStream/Smoothing.h"
+#include "Frame/Frame.h"
+#include "Frame/VectorFieldCalculator.h"
+#include "Session/Session.h"
+#include "Util/Message.h"
+
+static const std::string IMAGE_SHAPE = "1110 1110 25 4";
+static const std::string IMAGE_OPTS = "-s 0";
+static const std::string IMAGE_OPTS_NAN = "-s 0 -n row column -d 10";
+
+class TestFrame : public Frame {
+public:
+    TestFrame(uint32_t session_id, std::shared_ptr<carta::FileLoader> loader, const std::string& hdu, int default_z = DEFAULT_Z)
+        : Frame(session_id, loader, hdu, default_z) {}
+    FRIEND_TEST(VectorFieldTest, ExampleFriendTest);
```

Comment:
This seems to be copied from another file. There is no `ExampleFriendTest` in this test.